{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Concatenating all tools and strains at threshold 0, 10, 100, 1000, and 10K"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import allel\n",
    "import pandas as pd\n",
    "\n",
    "\n",
    "strain=['A_J','AKR_J','BALB_CJ','CBA_J','C3H_HeJ','DBA_2J','LP_J']\n",
    "\n",
    "tools=['BioGraph*',\n",
    "'breakdancer',\n",
    "'clever',\n",
    "'delly',\n",
    "'gasv',\n",
    "'gridss',\n",
    "'indelminer',\n",
    "'mistrvar',\n",
    "'pindel',\n",
    "'popdel',\n",
    "'rdxplorer',\n",
    "'smoove',\n",
    "'crest',\n",
    "'genomestrip',\n",
    "'manta_diploidSV',\n",
    "'Tardis']\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 0 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vc/.local/lib/python3.5/site-packages/allel/io/vcf_read.py:1736: UserWarning: multiple INFO headers for 'SVLEN'\n",
      "  warnings.warn('multiple INFO headers for %r' % k)\n",
      "/home/vc/.local/lib/python3.5/site-packages/ipykernel_launcher.py:13: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  del sys.path[0]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FP</td>\n",
       "      <td>389</td>\n",
       "      <td>3112809</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FP</td>\n",
       "      <td>1705</td>\n",
       "      <td>6055228</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FP</td>\n",
       "      <td>2445</td>\n",
       "      <td>6055227</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FP</td>\n",
       "      <td>582</td>\n",
       "      <td>6057082</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FP</td>\n",
       "      <td>140</td>\n",
       "      <td>6829728</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  flag length position strain threshold    tool\n",
       "0   FP    389  3112809   LP_J         0  Tardis\n",
       "1   FP   1705  6055228   LP_J         0  Tardis\n",
       "2   FP   2445  6055227   LP_J         0  Tardis\n",
       "3   FP    582  6057082   LP_J         0  Tardis\n",
       "4   FP    140  6829728   LP_J         0  Tardis"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#0 threshold deletions\n",
    "\n",
    "th=0\n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_0 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../Data/raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_0 = pd.concat([df_current, df_0],ignore_index=True)   \n",
    "        \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "            \n",
    "df_0.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vc/.local/lib/python3.5/site-packages/ipykernel_launcher.py:16: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  app.launch_new_instance()\n",
      "/home/vc/.local/lib/python3.5/site-packages/allel/io/vcf_read.py:322: UserWarning: error parsing integer value; field: POS; variant: -1\n",
      "  chunks = [d[0] for d in it]\n",
      "/home/vc/.local/lib/python3.5/site-packages/allel/io/vcf_read.py:322: UserWarning: error parsing integer value; field: POS; variant: 0\n",
      "  chunks = [d[0] for d in it]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TN</td>\n",
       "      <td>2942031</td>\n",
       "      <td>3113197</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TN</td>\n",
       "      <td>7962541</td>\n",
       "      <td>8890415</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TN</td>\n",
       "      <td>164720</td>\n",
       "      <td>16853151</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  flag   length  position strain threshold    tool\n",
       "0            -1         0   LP_J         0  Tardis\n",
       "1            -1         0   LP_J         0  Tardis\n",
       "2   TN  2942031   3113197   LP_J         0  Tardis\n",
       "3   TN  7962541   8890415   LP_J         0  Tardis\n",
       "4   TN   164720  16853151   LP_J         0  Tardis"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#0 threshold non-deletions\n",
    "import os\n",
    "\n",
    "th=0\n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_0_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        \n",
    "        file='../Data/raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "#             print (\"----\")\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_0_nondel = pd.concat([df_current, df_0_nondel],ignore_index=True)   \n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "        \n",
    "df_0_nondel.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 10 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vc/.local/lib/python3.5/site-packages/ipykernel_launcher.py:17: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FP</td>\n",
       "      <td>389</td>\n",
       "      <td>3112809</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FP</td>\n",
       "      <td>1705</td>\n",
       "      <td>6055228</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FP</td>\n",
       "      <td>2445</td>\n",
       "      <td>6055227</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FP</td>\n",
       "      <td>582</td>\n",
       "      <td>6057082</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FP</td>\n",
       "      <td>140</td>\n",
       "      <td>6829728</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  flag length position strain threshold    tool\n",
       "0   FP    389  3112809   LP_J        10  Tardis\n",
       "1   FP   1705  6055228   LP_J        10  Tardis\n",
       "2   FP   2445  6055227   LP_J        10  Tardis\n",
       "3   FP    582  6057082   LP_J        10  Tardis\n",
       "4   FP    140  6829728   LP_J        10  Tardis"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#10 threshold deletions\n",
    "#missing files: pindel.BALB_CJ, and platypus.BALB_CJ\n",
    "#tools were not able to run on mouse data; ignore?\n",
    "\n",
    "th=10     \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_10 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../Data/raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        if os.path.exists(file):\n",
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_10 = pd.concat([df_current, df_10],ignore_index=True)\n",
    "\n",
    "            else:\n",
    "                print(\"Empty File: \" + file)\n",
    "        else:\n",
    "            print(\"Nonexistant File: \" + file)\n",
    "            \n",
    "df_10.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vc/.local/lib/python3.5/site-packages/ipykernel_launcher.py:13: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  del sys.path[0]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TN</td>\n",
       "      <td>2942031</td>\n",
       "      <td>3113197</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TN</td>\n",
       "      <td>7962541</td>\n",
       "      <td>8890415</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TN</td>\n",
       "      <td>164720</td>\n",
       "      <td>16853151</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  flag   length  position strain threshold    tool\n",
       "0            -1         0   LP_J        10  Tardis\n",
       "1            -1         0   LP_J        10  Tardis\n",
       "2   TN  2942031   3113197   LP_J        10  Tardis\n",
       "3   TN  7962541   8890415   LP_J        10  Tardis\n",
       "4   TN   164720  16853151   LP_J        10  Tardis"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#10 threshold non-deletions\n",
    "th=10     \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_10_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../Data/raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_10_nondel = pd.concat([df_current, df_10_nondel],ignore_index=True)\n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "            \n",
    "df_10_nondel.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 100 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vc/.local/lib/python3.5/site-packages/ipykernel_launcher.py:16: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  app.launch_new_instance()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FP</td>\n",
       "      <td>389</td>\n",
       "      <td>3112809</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FP</td>\n",
       "      <td>1705</td>\n",
       "      <td>6055228</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FP</td>\n",
       "      <td>582</td>\n",
       "      <td>6057082</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FP</td>\n",
       "      <td>140</td>\n",
       "      <td>6829728</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FP</td>\n",
       "      <td>1922</td>\n",
       "      <td>7215484</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  flag length position strain threshold    tool\n",
       "0   FP    389  3112809   LP_J       100  Tardis\n",
       "1   FP   1705  6055228   LP_J       100  Tardis\n",
       "2   FP    582  6057082   LP_J       100  Tardis\n",
       "3   FP    140  6829728   LP_J       100  Tardis\n",
       "4   FP   1922  7215484   LP_J       100  Tardis"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#100 threshold deletions\n",
    "#missing files: pindel.BALB_CJ, and platypus.BALB_CJ\n",
    "\n",
    "th=100\n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_100 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../Data/raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        if os.path.exists(file):\n",
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_100 = pd.concat([df_current, df_100],ignore_index=True)\n",
    "        \n",
    "            else:\n",
    "                print(\"Empty File: \" + file)\n",
    "                \n",
    "        else:\n",
    "            print(\"Nonexistant File: \" + file)\n",
    "            \n",
    "df_100.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vc/.local/lib/python3.5/site-packages/ipykernel_launcher.py:13: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  del sys.path[0]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TN</td>\n",
       "      <td>2942031</td>\n",
       "      <td>3113197</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TN</td>\n",
       "      <td>7962541</td>\n",
       "      <td>8890415</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TN</td>\n",
       "      <td>164720</td>\n",
       "      <td>16853151</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  flag   length  position strain threshold    tool\n",
       "0            -1         0   LP_J       100  Tardis\n",
       "1            -1         0   LP_J       100  Tardis\n",
       "2   TN  2942031   3113197   LP_J       100  Tardis\n",
       "3   TN  7962541   8890415   LP_J       100  Tardis\n",
       "4   TN   164720  16853151   LP_J       100  Tardis"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#100 threshold non-deletions\n",
    "\n",
    "th=100     \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_100_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../Data/raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_100_nondel = pd.concat([df_current, df_100_nondel],ignore_index=True)\n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "            \n",
    "df_100_nondel.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1000 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vc/.local/lib/python3.5/site-packages/ipykernel_launcher.py:16: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  app.launch_new_instance()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nonexistant File: ../Data/raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.Tardis.A_J.chr19.100p_sorted.modified.vcf\n",
      "Nonexistant File: ../Data/raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.Tardis.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "Nonexistant File: ../Data/raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.Tardis.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "Nonexistant File: ../Data/raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.Tardis.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "Nonexistant File: ../Data/raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.Tardis.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "Nonexistant File: ../Data/raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.Tardis.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "Nonexistant File: ../Data/raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.Tardis.LP_J.chr19.100p_sorted.modified.vcf\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FP</td>\n",
       "      <td>389</td>\n",
       "      <td>3112809</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FP</td>\n",
       "      <td>2445</td>\n",
       "      <td>6055227</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FP</td>\n",
       "      <td>582</td>\n",
       "      <td>6057082</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FP</td>\n",
       "      <td>140</td>\n",
       "      <td>6829728</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FP</td>\n",
       "      <td>1922</td>\n",
       "      <td>7215484</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  flag length position strain threshold    tool\n",
       "0   FP    389  3112809   LP_J      1000  Tardis\n",
       "1   FP   2445  6055227   LP_J      1000  Tardis\n",
       "2   FP    582  6057082   LP_J      1000  Tardis\n",
       "3   FP    140  6829728   LP_J      1000  Tardis\n",
       "4   FP   1922  7215484   LP_J      1000  Tardis"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#1000 threshold deletions\n",
    "#missing files: pindel.BALB_CJ, and platypus.BALB_CJ\n",
    "\n",
    "th=1000    \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_1000 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../Data/raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        if os.path.exists(file):\n",
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_1000 = pd.concat([df_current, df_1000],ignore_index=True)\n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "            \n",
    "    else:\n",
    "        print (\"Nonexistant File: \" + file)\n",
    "           \n",
    "df_1000.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FP</td>\n",
       "      <td>389</td>\n",
       "      <td>3112809</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FP</td>\n",
       "      <td>2445</td>\n",
       "      <td>6055227</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FP</td>\n",
       "      <td>582</td>\n",
       "      <td>6057082</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FP</td>\n",
       "      <td>140</td>\n",
       "      <td>6829728</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FP</td>\n",
       "      <td>1922</td>\n",
       "      <td>7215484</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>FP</td>\n",
       "      <td>311</td>\n",
       "      <td>17277456</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>FP</td>\n",
       "      <td>6219</td>\n",
       "      <td>17303364</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>FP</td>\n",
       "      <td>546</td>\n",
       "      <td>17335461</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>FP</td>\n",
       "      <td>20749</td>\n",
       "      <td>21598360</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>FP</td>\n",
       "      <td>33386</td>\n",
       "      <td>21619242</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>FP</td>\n",
       "      <td>1258</td>\n",
       "      <td>23208995</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>FP</td>\n",
       "      <td>3434</td>\n",
       "      <td>23210630</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>FP</td>\n",
       "      <td>3120</td>\n",
       "      <td>23210630</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>FP</td>\n",
       "      <td>4208</td>\n",
       "      <td>23214196</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>FP</td>\n",
       "      <td>489</td>\n",
       "      <td>23214152</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>FP</td>\n",
       "      <td>3645</td>\n",
       "      <td>23214767</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>FP</td>\n",
       "      <td>10369</td>\n",
       "      <td>23218555</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>FP</td>\n",
       "      <td>9627</td>\n",
       "      <td>23218555</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>FP</td>\n",
       "      <td>629</td>\n",
       "      <td>23228307</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>FP</td>\n",
       "      <td>347</td>\n",
       "      <td>25958129</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>FP</td>\n",
       "      <td>578</td>\n",
       "      <td>27095140</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>FP</td>\n",
       "      <td>302</td>\n",
       "      <td>39692745</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>FP</td>\n",
       "      <td>4820</td>\n",
       "      <td>41468141</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>FP</td>\n",
       "      <td>401</td>\n",
       "      <td>42939428</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>FP</td>\n",
       "      <td>744</td>\n",
       "      <td>43536647</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>FP</td>\n",
       "      <td>708</td>\n",
       "      <td>44186087</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>FP</td>\n",
       "      <td>161</td>\n",
       "      <td>46048741</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>FP</td>\n",
       "      <td>689</td>\n",
       "      <td>52759777</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>FP</td>\n",
       "      <td>255</td>\n",
       "      <td>53349176</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>FP</td>\n",
       "      <td>336</td>\n",
       "      <td>55518257</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864323</th>\n",
       "      <td>TP</td>\n",
       "      <td>130</td>\n",
       "      <td>59977556</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864324</th>\n",
       "      <td>TP</td>\n",
       "      <td>135</td>\n",
       "      <td>59983358</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864325</th>\n",
       "      <td>TP</td>\n",
       "      <td>1527</td>\n",
       "      <td>59998319</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864326</th>\n",
       "      <td>TP</td>\n",
       "      <td>6245</td>\n",
       "      <td>60018700</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864327</th>\n",
       "      <td>TP</td>\n",
       "      <td>636</td>\n",
       "      <td>60056188</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864328</th>\n",
       "      <td>TP</td>\n",
       "      <td>212</td>\n",
       "      <td>60070105</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864329</th>\n",
       "      <td>TP</td>\n",
       "      <td>162</td>\n",
       "      <td>60231982</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864330</th>\n",
       "      <td>TP</td>\n",
       "      <td>102</td>\n",
       "      <td>60241348</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864331</th>\n",
       "      <td>TP</td>\n",
       "      <td>201</td>\n",
       "      <td>60290957</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864332</th>\n",
       "      <td>TP</td>\n",
       "      <td>344</td>\n",
       "      <td>60309278</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864333</th>\n",
       "      <td>TP</td>\n",
       "      <td>7227</td>\n",
       "      <td>60347559</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864334</th>\n",
       "      <td>TP</td>\n",
       "      <td>91</td>\n",
       "      <td>60479695</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864335</th>\n",
       "      <td>TP</td>\n",
       "      <td>221</td>\n",
       "      <td>60572725</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864336</th>\n",
       "      <td>TP</td>\n",
       "      <td>215</td>\n",
       "      <td>60585844</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864337</th>\n",
       "      <td>TP</td>\n",
       "      <td>181</td>\n",
       "      <td>60627412</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864338</th>\n",
       "      <td>TP</td>\n",
       "      <td>1082</td>\n",
       "      <td>60641256</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864339</th>\n",
       "      <td>TP</td>\n",
       "      <td>6478</td>\n",
       "      <td>60643753</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864340</th>\n",
       "      <td>TP</td>\n",
       "      <td>540</td>\n",
       "      <td>60657081</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864341</th>\n",
       "      <td>TP</td>\n",
       "      <td>96</td>\n",
       "      <td>60690742</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864342</th>\n",
       "      <td>TP</td>\n",
       "      <td>1133</td>\n",
       "      <td>60703643</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864343</th>\n",
       "      <td>TP</td>\n",
       "      <td>126</td>\n",
       "      <td>60744211</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864344</th>\n",
       "      <td>TP</td>\n",
       "      <td>146</td>\n",
       "      <td>60801537</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864345</th>\n",
       "      <td>TP</td>\n",
       "      <td>692</td>\n",
       "      <td>60828815</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864346</th>\n",
       "      <td>TP</td>\n",
       "      <td>134</td>\n",
       "      <td>60852991</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864347</th>\n",
       "      <td>TP</td>\n",
       "      <td>8731</td>\n",
       "      <td>60925302</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864348</th>\n",
       "      <td>TP</td>\n",
       "      <td>156</td>\n",
       "      <td>60948603</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864349</th>\n",
       "      <td>TP</td>\n",
       "      <td>203</td>\n",
       "      <td>61026660</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864350</th>\n",
       "      <td>TP</td>\n",
       "      <td>609</td>\n",
       "      <td>61114188</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864351</th>\n",
       "      <td>TP</td>\n",
       "      <td>6468</td>\n",
       "      <td>61168770</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864352</th>\n",
       "      <td>TP</td>\n",
       "      <td>121</td>\n",
       "      <td>61192223</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>BioGraph*</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>864353 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       flag length  position strain threshold       tool\n",
       "0        FP    389   3112809   LP_J      1000     Tardis\n",
       "1        FP   2445   6055227   LP_J      1000     Tardis\n",
       "2        FP    582   6057082   LP_J      1000     Tardis\n",
       "3        FP    140   6829728   LP_J      1000     Tardis\n",
       "4        FP   1922   7215484   LP_J      1000     Tardis\n",
       "5        FP    311  17277456   LP_J      1000     Tardis\n",
       "6        FP   6219  17303364   LP_J      1000     Tardis\n",
       "7        FP    546  17335461   LP_J      1000     Tardis\n",
       "8        FP  20749  21598360   LP_J      1000     Tardis\n",
       "9        FP  33386  21619242   LP_J      1000     Tardis\n",
       "10       FP   1258  23208995   LP_J      1000     Tardis\n",
       "11       FP   3434  23210630   LP_J      1000     Tardis\n",
       "12       FP   3120  23210630   LP_J      1000     Tardis\n",
       "13       FP   4208  23214196   LP_J      1000     Tardis\n",
       "14       FP    489  23214152   LP_J      1000     Tardis\n",
       "15       FP   3645  23214767   LP_J      1000     Tardis\n",
       "16       FP  10369  23218555   LP_J      1000     Tardis\n",
       "17       FP   9627  23218555   LP_J      1000     Tardis\n",
       "18       FP    629  23228307   LP_J      1000     Tardis\n",
       "19       FP    347  25958129   LP_J      1000     Tardis\n",
       "20       FP    578  27095140   LP_J      1000     Tardis\n",
       "21       FP    302  39692745   LP_J      1000     Tardis\n",
       "22       FP   4820  41468141   LP_J      1000     Tardis\n",
       "23       FP    401  42939428   LP_J      1000     Tardis\n",
       "24       FP    744  43536647   LP_J      1000     Tardis\n",
       "25       FP    708  44186087   LP_J      1000     Tardis\n",
       "26       FP    161  46048741   LP_J      1000     Tardis\n",
       "27       FP    689  52759777   LP_J      1000     Tardis\n",
       "28       FP    255  53349176   LP_J      1000     Tardis\n",
       "29       FP    336  55518257   LP_J      1000     Tardis\n",
       "...     ...    ...       ...    ...       ...        ...\n",
       "864323   TP    130  59977556    A_J      1000  BioGraph*\n",
       "864324   TP    135  59983358    A_J      1000  BioGraph*\n",
       "864325   TP   1527  59998319    A_J      1000  BioGraph*\n",
       "864326   TP   6245  60018700    A_J      1000  BioGraph*\n",
       "864327   TP    636  60056188    A_J      1000  BioGraph*\n",
       "864328   TP    212  60070105    A_J      1000  BioGraph*\n",
       "864329   TP    162  60231982    A_J      1000  BioGraph*\n",
       "864330   TP    102  60241348    A_J      1000  BioGraph*\n",
       "864331   TP    201  60290957    A_J      1000  BioGraph*\n",
       "864332   TP    344  60309278    A_J      1000  BioGraph*\n",
       "864333   TP   7227  60347559    A_J      1000  BioGraph*\n",
       "864334   TP     91  60479695    A_J      1000  BioGraph*\n",
       "864335   TP    221  60572725    A_J      1000  BioGraph*\n",
       "864336   TP    215  60585844    A_J      1000  BioGraph*\n",
       "864337   TP    181  60627412    A_J      1000  BioGraph*\n",
       "864338   TP   1082  60641256    A_J      1000  BioGraph*\n",
       "864339   TP   6478  60643753    A_J      1000  BioGraph*\n",
       "864340   TP    540  60657081    A_J      1000  BioGraph*\n",
       "864341   TP     96  60690742    A_J      1000  BioGraph*\n",
       "864342   TP   1133  60703643    A_J      1000  BioGraph*\n",
       "864343   TP    126  60744211    A_J      1000  BioGraph*\n",
       "864344   TP    146  60801537    A_J      1000  BioGraph*\n",
       "864345   TP    692  60828815    A_J      1000  BioGraph*\n",
       "864346   TP    134  60852991    A_J      1000  BioGraph*\n",
       "864347   TP   8731  60925302    A_J      1000  BioGraph*\n",
       "864348   TP    156  60948603    A_J      1000  BioGraph*\n",
       "864349   TP    203  61026660    A_J      1000  BioGraph*\n",
       "864350   TP    609  61114188    A_J      1000  BioGraph*\n",
       "864351   TP   6468  61168770    A_J      1000  BioGraph*\n",
       "864352   TP    121  61192223    A_J      1000  BioGraph*\n",
       "\n",
       "[864353 rows x 6 columns]"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vc/.local/lib/python3.5/site-packages/ipykernel_launcher.py:15: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  from ipykernel import kernelapp as app\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TN</td>\n",
       "      <td>2942031</td>\n",
       "      <td>3113197</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TN</td>\n",
       "      <td>7962541</td>\n",
       "      <td>8890415</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TN</td>\n",
       "      <td>164720</td>\n",
       "      <td>16853151</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  flag   length  position strain threshold    tool\n",
       "0            -1         0   LP_J      1000  Tardis\n",
       "1            -1         0   LP_J      1000  Tardis\n",
       "2   TN  2942031   3113197   LP_J      1000  Tardis\n",
       "3   TN  7962541   8890415   LP_J      1000  Tardis\n",
       "4   TN   164720  16853151   LP_J      1000  Tardis"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#1000 threshold non-deletions\n",
    "\n",
    "th=1000    \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_1000_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../Data/raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
    "        \n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_1000_nondel = pd.concat([df_current, df_1000_nondel],ignore_index=True)\n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "        \n",
    "df_1000_nondel.head()\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 10000 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vc/.local/lib/python3.5/site-packages/ipykernel_launcher.py:16: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  app.launch_new_instance()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FP</td>\n",
       "      <td>389</td>\n",
       "      <td>3112809</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FP</td>\n",
       "      <td>2445</td>\n",
       "      <td>6055227</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FP</td>\n",
       "      <td>582</td>\n",
       "      <td>6057082</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FP</td>\n",
       "      <td>140</td>\n",
       "      <td>6829728</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FP</td>\n",
       "      <td>1922</td>\n",
       "      <td>7215484</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  flag length position strain threshold    tool\n",
       "0   FP    389  3112809   LP_J     10000  Tardis\n",
       "1   FP   2445  6055227   LP_J     10000  Tardis\n",
       "2   FP    582  6057082   LP_J     10000  Tardis\n",
       "3   FP    140  6829728   LP_J     10000  Tardis\n",
       "4   FP   1922  7215484   LP_J     10000  Tardis"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#10000 threshold deletions\n",
    "#missing files: pindel.BALB_CJ, and platypus.BALB_CJ\n",
    "\n",
    "th=10000    \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_10000 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../Data/raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        if os.path.exists(file):\n",
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_10000 = pd.concat([df_current, df_10000],ignore_index=True)\n",
    "            \n",
    "            else:\n",
    "                print(\"Empty File: \" + file)\n",
    "                \n",
    "        else:\n",
    "            print(\"Nonexistant File: \" + file)\n",
    "        \n",
    "df_10000.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vc/.local/lib/python3.5/site-packages/ipykernel_launcher.py:15: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  from ipykernel import kernelapp as app\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TN</td>\n",
       "      <td>2942031</td>\n",
       "      <td>3113197</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TN</td>\n",
       "      <td>7962541</td>\n",
       "      <td>8890415</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TN</td>\n",
       "      <td>164720</td>\n",
       "      <td>16853151</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  flag   length  position strain threshold    tool\n",
       "0            -1         0   LP_J     10000  Tardis\n",
       "1            -1         0   LP_J     10000  Tardis\n",
       "2   TN  2942031   3113197   LP_J     10000  Tardis\n",
       "3   TN  7962541   8890415   LP_J     10000  Tardis\n",
       "4   TN   164720  16853151   LP_J     10000  Tardis"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#10000 threshold non-deletions\n",
    "\n",
    "th=10000    \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_10000_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../Data/raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
    "        \n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_10000_nondel = pd.concat([df_current, df_10000_nondel],ignore_index=True)\n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "           \n",
    "        \n",
    "df_10000_nondel.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Stored 'df_10' (DataFrame)\n",
      "Stored 'df_100' (DataFrame)\n",
      "Stored 'df_1000' (DataFrame)\n",
      "Stored 'df_10000' (DataFrame)\n",
      "Stored 'df_10_nondel' (DataFrame)\n",
      "Stored 'df_100_nondel' (DataFrame)\n",
      "Stored 'df_1000_nondel' (DataFrame)\n",
      "Stored 'df_10000_nondel' (DataFrame)\n"
     ]
    }
   ],
   "source": [
    "%store df_10\n",
    "%store df_100\n",
    "%store df_1000\n",
    "%store df_10000\n",
    "%store df_10_nondel\n",
    "%store df_100_nondel\n",
    "%store df_1000_nondel\n",
    "%store df_10000_nondel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_10.to_csv('../Data/analysis_files/df_10.csv')\n",
    "df_100.to_csv('../Data/analysis_files/df_100.csv')\n",
    "df_1000.to_csv('../Data/analysis_files/df_1000.csv')\n",
    "df_10000.to_csv('../Data/analysis_files/df_10000.csv')\n",
    "\n",
    "df_10_nondel.to_csv('../Data/analysis_files/df_10_nondel.csv')\n",
    "df_100_nondel.to_csv('../Data/analysis_files/df_100_nondel.csv')\n",
    "df_1000_nondel.to_csv('../Data/analysis_files/df_1000_nondel.csv')\n",
    "df_10000_nondel.to_csv('../Data/analysis_files/df_10000_nondel.csv')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Concatenate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "#List of your dataframes\n",
    "#removed df_0 and df_0_nondel for graphing purposes\n",
    "pdList = [df_10,df_100,df_1000,df_10000,df_10_nondel,df_100_nondel,df_1000_nondel,df_10000_nondel] \n",
    "df_cat = pd.concat(pdList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FP</td>\n",
       "      <td>389</td>\n",
       "      <td>3112809</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FP</td>\n",
       "      <td>1705</td>\n",
       "      <td>6055228</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FP</td>\n",
       "      <td>2445</td>\n",
       "      <td>6055227</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FP</td>\n",
       "      <td>582</td>\n",
       "      <td>6057082</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FP</td>\n",
       "      <td>140</td>\n",
       "      <td>6829728</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>Tardis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  flag length position strain threshold    tool\n",
       "0   FP    389  3112809   LP_J        10  Tardis\n",
       "1   FP   1705  6055228   LP_J        10  Tardis\n",
       "2   FP   2445  6055227   LP_J        10  Tardis\n",
       "3   FP    582  6057082   LP_J        10  Tardis\n",
       "4   FP    140  6829728   LP_J        10  Tardis"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cat.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(434, 4)"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#output counts for TP, FP, TN rows are not the same as Varuni's; due to missing files?\n",
    "group_data_TP=df_cat[df_cat['flag'] == 'TP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_data_TP=group_data_TP.rename(columns={\"flag\": \"nTP\"})\n",
    "group_data_TP.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>A_J</td>\n",
       "      <td>279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>A_J</td>\n",
       "      <td>304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>316</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         tool  threshold   strain  nTP\n",
       "0   BioGraph*         10    AKR_J   36\n",
       "1   BioGraph*         10      A_J   47\n",
       "2   BioGraph*         10  BALB_CJ   45\n",
       "3   BioGraph*         10  C3H_HeJ   41\n",
       "4   BioGraph*         10    CBA_J   44\n",
       "5   BioGraph*         10   DBA_2J   45\n",
       "6   BioGraph*         10     LP_J   39\n",
       "7   BioGraph*        100    AKR_J  249\n",
       "8   BioGraph*        100      A_J  279\n",
       "9   BioGraph*        100  BALB_CJ  269\n",
       "10  BioGraph*        100  C3H_HeJ  251\n",
       "11  BioGraph*        100    CBA_J  272\n",
       "12  BioGraph*        100   DBA_2J  285\n",
       "13  BioGraph*        100     LP_J  222\n",
       "14  BioGraph*       1000    AKR_J  269\n",
       "15  BioGraph*       1000      A_J  304\n",
       "16  BioGraph*       1000  BALB_CJ  294\n",
       "17  BioGraph*       1000  C3H_HeJ  278\n",
       "18  BioGraph*       1000    CBA_J  302\n",
       "19  BioGraph*       1000   DBA_2J  316"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_TP.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(448, 4)"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_FP=df_cat[df_cat['flag'] == 'FP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_data_FP=group_data_FP.rename(columns={\"flag\": \"nFP\"})\n",
    "group_data_FP.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nFP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>445</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nFP\n",
       "0  BioGraph*         10    AKR_J  389\n",
       "1  BioGraph*         10      A_J  421\n",
       "2  BioGraph*         10  BALB_CJ  417\n",
       "3  BioGraph*         10  C3H_HeJ  416\n",
       "4  BioGraph*         10    CBA_J  445"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_FP.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(448, 4)"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_TN=df_cat[df_cat['flag'] == 'TN'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_data_TN=group_data_TN.rename(columns={\"flag\": \"nTN\"})\n",
    "group_data_TN.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(448, 4)"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#fill missing combinations with 0s\n",
    "#removed 0 threshold group since it = 0 for all tools\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_data_TP['tool'] == t) & (group_data_TP['strain'] == s) & (group_data_TP['threshold'] == th)).any()):\n",
    "                group_data_TP = group_data_TP.append({'tool' : t , 'strain' : s,'threshold' : th,'nTP' : 0} , ignore_index=True)\n",
    "group_data_TP.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(448, 4)"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_data_FP['tool'] == t) & (group_data_FP['strain'] == s) & (group_data_FP['threshold'] == th)).any()):\n",
    "                #group_data_FP = group_data_FP.append({'tool' : t , 'strain' : s,'threshold' : th,'nFP' : 0} , ignore_index=True)\n",
    "                print (s,t,th)\n",
    "group_data_FP.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(448, 4)"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_data_TN['tool'] == t) & (group_data_TN['strain'] == s) & (group_data_TN['threshold'] == th)).any()):\n",
    "                group_data_TN = group_data_TN.append({'tool' : t , 'strain' : s,'threshold' : th,'nTN' : 0} , ignore_index=True)\n",
    "                #print (s,t,th)\n",
    "group_data_TN.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(448, 6)"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#merge TP and FP\n",
    "df_merge=pd.merge(group_data_TP, group_data_FP,on=['tool','threshold','strain'])\n",
    "df_merge=pd.merge(df_merge, group_data_TN,on=['tool','threshold','strain'])\n",
    "df_merge.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>36</td>\n",
       "      <td>389</td>\n",
       "      <td>206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>47</td>\n",
       "      <td>421</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>45</td>\n",
       "      <td>417</td>\n",
       "      <td>237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>41</td>\n",
       "      <td>416</td>\n",
       "      <td>247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>44</td>\n",
       "      <td>445</td>\n",
       "      <td>270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>45</td>\n",
       "      <td>454</td>\n",
       "      <td>279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>39</td>\n",
       "      <td>336</td>\n",
       "      <td>213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>249</td>\n",
       "      <td>176</td>\n",
       "      <td>289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>A_J</td>\n",
       "      <td>279</td>\n",
       "      <td>189</td>\n",
       "      <td>317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>269</td>\n",
       "      <td>193</td>\n",
       "      <td>330</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP  nFP  nTN\n",
       "0  BioGraph*         10    AKR_J   36  389  206\n",
       "1  BioGraph*         10      A_J   47  421  220\n",
       "2  BioGraph*         10  BALB_CJ   45  417  237\n",
       "3  BioGraph*         10  C3H_HeJ   41  416  247\n",
       "4  BioGraph*         10    CBA_J   44  445  270\n",
       "5  BioGraph*         10   DBA_2J   45  454  279\n",
       "6  BioGraph*         10     LP_J   39  336  213\n",
       "7  BioGraph*        100    AKR_J  249  176  289\n",
       "8  BioGraph*        100      A_J  279  189  317\n",
       "9  BioGraph*        100  BALB_CJ  269  193  330"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vc/.local/lib/python3.5/site-packages/ipykernel_launcher.py:11: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  # This is added back by InteractiveShellApp.init_path()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain</th>\n",
       "      <th>n_true</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKR_J</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A_J</td>\n",
       "      <td>533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBA_J</td>\n",
       "      <td>586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>LP_J</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    strain  n_true\n",
       "0    AKR_J     504\n",
       "1      A_J     533\n",
       "2  BALB_CJ     545\n",
       "3  C3H_HeJ     539\n",
       "4    CBA_J     586\n",
       "5   DBA_2J     609\n",
       "6     LP_J     483"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#true SV deletions\n",
    "df_current = pd.DataFrame(columns=['strain','length'])\n",
    "df = pd.DataFrame(columns=['strain','length'])\n",
    "\n",
    "\n",
    "for s in strain:\n",
    "    file='../Data/gold_standard/mouse_vcf/'+s+'_reference.vcf'\n",
    "    callset = allel.read_vcf(file,fields='*')\n",
    "    \n",
    "    df_current = pd.DataFrame({'strain': s, 'length': callset['variants/SVLEN']})\n",
    "    df = pd.concat([df_current, df],ignore_index=True)\n",
    "group_data_true = df.groupby(['strain'],as_index=False).count()\n",
    "group_data_true=group_data_true.rename(columns={\"length\": \"n_true\"})\n",
    "group_data_true"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(448, 7)"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge=pd.merge(df_merge, group_data_true)\n",
    "df_merge.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>n_true</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>36</td>\n",
       "      <td>389</td>\n",
       "      <td>206</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>249</td>\n",
       "      <td>176</td>\n",
       "      <td>289</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>269</td>\n",
       "      <td>156</td>\n",
       "      <td>299</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>291</td>\n",
       "      <td>134</td>\n",
       "      <td>347</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Tardis</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>51</td>\n",
       "      <td>245</td>\n",
       "      <td>284</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Tardis</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>217</td>\n",
       "      <td>79</td>\n",
       "      <td>341</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Tardis</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>243</td>\n",
       "      <td>53</td>\n",
       "      <td>350</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Tardis</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>250</td>\n",
       "      <td>46</td>\n",
       "      <td>390</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>9</td>\n",
       "      <td>527</td>\n",
       "      <td>253</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>254</td>\n",
       "      <td>282</td>\n",
       "      <td>314</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>311</td>\n",
       "      <td>225</td>\n",
       "      <td>348</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>325</td>\n",
       "      <td>211</td>\n",
       "      <td>393</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>clever</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>54</td>\n",
       "      <td>9551</td>\n",
       "      <td>31</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>clever</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>336</td>\n",
       "      <td>9269</td>\n",
       "      <td>53</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>clever</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>396</td>\n",
       "      <td>9209</td>\n",
       "      <td>188</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>clever</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>488</td>\n",
       "      <td>9117</td>\n",
       "      <td>298</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>crest</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>67</td>\n",
       "      <td>159</td>\n",
       "      <td>1007</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>crest</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>138</td>\n",
       "      <td>88</td>\n",
       "      <td>1015</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>crest</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>149</td>\n",
       "      <td>77</td>\n",
       "      <td>1099</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>delly</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>69</td>\n",
       "      <td>854</td>\n",
       "      <td>117</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>delly</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>311</td>\n",
       "      <td>612</td>\n",
       "      <td>188</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>delly</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>353</td>\n",
       "      <td>570</td>\n",
       "      <td>208</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>delly</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>405</td>\n",
       "      <td>518</td>\n",
       "      <td>379</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>gasv</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>68</td>\n",
       "      <td>12328</td>\n",
       "      <td>9</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>gasv</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>304</td>\n",
       "      <td>12092</td>\n",
       "      <td>35</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>gasv</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>496</td>\n",
       "      <td>11900</td>\n",
       "      <td>508</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>genomestrip</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>12</td>\n",
       "      <td>218</td>\n",
       "      <td>116</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>genomestrip</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>126</td>\n",
       "      <td>104</td>\n",
       "      <td>170</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>genomestrip</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>148</td>\n",
       "      <td>82</td>\n",
       "      <td>188</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>genomestrip</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>156</td>\n",
       "      <td>74</td>\n",
       "      <td>298</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>indelminer</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>15</td>\n",
       "      <td>13401</td>\n",
       "      <td>137</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>indelminer</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>67</td>\n",
       "      <td>13349</td>\n",
       "      <td>137</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>indelminer</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>189</td>\n",
       "      <td>13227</td>\n",
       "      <td>159</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>indelminer</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>347</td>\n",
       "      <td>13069</td>\n",
       "      <td>1265</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>manta_diploidSV</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>60</td>\n",
       "      <td>422</td>\n",
       "      <td>873</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>manta_diploidSV</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>312</td>\n",
       "      <td>170</td>\n",
       "      <td>1052</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>manta_diploidSV</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>337</td>\n",
       "      <td>145</td>\n",
       "      <td>1077</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>manta_diploidSV</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>361</td>\n",
       "      <td>121</td>\n",
       "      <td>1293</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>426</th>\n",
       "      <td>mistrvar</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>18</td>\n",
       "      <td>5554</td>\n",
       "      <td>54</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>427</th>\n",
       "      <td>mistrvar</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>97</td>\n",
       "      <td>5475</td>\n",
       "      <td>59</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>mistrvar</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>183</td>\n",
       "      <td>5389</td>\n",
       "      <td>82</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>mistrvar</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>430</td>\n",
       "      <td>5142</td>\n",
       "      <td>789</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>pindel</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>43</td>\n",
       "      <td>76321</td>\n",
       "      <td>43</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>pindel</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>195</td>\n",
       "      <td>76169</td>\n",
       "      <td>45</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>pindel</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>340</td>\n",
       "      <td>76024</td>\n",
       "      <td>131</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>pindel</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>480</td>\n",
       "      <td>75884</td>\n",
       "      <td>4239</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>popdel</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>6</td>\n",
       "      <td>216</td>\n",
       "      <td>256</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>popdel</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>173</td>\n",
       "      <td>49</td>\n",
       "      <td>317</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>popdel</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>210</td>\n",
       "      <td>12</td>\n",
       "      <td>336</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>popdel</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>215</td>\n",
       "      <td>7</td>\n",
       "      <td>359</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>rdxplorer</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>4</td>\n",
       "      <td>2662</td>\n",
       "      <td>53</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>rdxplorer</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>132</td>\n",
       "      <td>2534</td>\n",
       "      <td>27</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>rdxplorer</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>227</td>\n",
       "      <td>2439</td>\n",
       "      <td>59</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>rdxplorer</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>393</td>\n",
       "      <td>2273</td>\n",
       "      <td>187</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>442</th>\n",
       "      <td>smoove</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>78</td>\n",
       "      <td>322</td>\n",
       "      <td>245</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>443</th>\n",
       "      <td>smoove</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>281</td>\n",
       "      <td>119</td>\n",
       "      <td>331</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>444</th>\n",
       "      <td>smoove</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>315</td>\n",
       "      <td>85</td>\n",
       "      <td>348</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>445</th>\n",
       "      <td>smoove</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>331</td>\n",
       "      <td>69</td>\n",
       "      <td>423</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>446</th>\n",
       "      <td>gasv</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>11605</td>\n",
       "      <td>9</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>447</th>\n",
       "      <td>crest</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>299</td>\n",
       "      <td>3517</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>448 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                tool  threshold strain  nTP    nFP   nTN  n_true\n",
       "0          BioGraph*         10  AKR_J   36    389   206     504\n",
       "1          BioGraph*        100  AKR_J  249    176   289     504\n",
       "2          BioGraph*       1000  AKR_J  269    156   299     504\n",
       "3          BioGraph*      10000  AKR_J  291    134   347     504\n",
       "4             Tardis         10  AKR_J   51    245   284     504\n",
       "5             Tardis        100  AKR_J  217     79   341     504\n",
       "6             Tardis       1000  AKR_J  243     53   350     504\n",
       "7             Tardis      10000  AKR_J  250     46   390     504\n",
       "8        breakdancer         10  AKR_J    9    527   253     504\n",
       "9        breakdancer        100  AKR_J  254    282   314     504\n",
       "10       breakdancer       1000  AKR_J  311    225   348     504\n",
       "11       breakdancer      10000  AKR_J  325    211   393     504\n",
       "12            clever         10  AKR_J   54   9551    31     504\n",
       "13            clever        100  AKR_J  336   9269    53     504\n",
       "14            clever       1000  AKR_J  396   9209   188     504\n",
       "15            clever      10000  AKR_J  488   9117   298     504\n",
       "16             crest        100  AKR_J   67    159  1007     504\n",
       "17             crest       1000  AKR_J  138     88  1015     504\n",
       "18             crest      10000  AKR_J  149     77  1099     504\n",
       "19             delly         10  AKR_J   69    854   117     504\n",
       "20             delly        100  AKR_J  311    612   188     504\n",
       "21             delly       1000  AKR_J  353    570   208     504\n",
       "22             delly      10000  AKR_J  405    518   379     504\n",
       "23              gasv        100  AKR_J   68  12328     9     504\n",
       "24              gasv       1000  AKR_J  304  12092    35     504\n",
       "25              gasv      10000  AKR_J  496  11900   508     504\n",
       "26       genomestrip         10  AKR_J   12    218   116     504\n",
       "27       genomestrip        100  AKR_J  126    104   170     504\n",
       "28       genomestrip       1000  AKR_J  148     82   188     504\n",
       "29       genomestrip      10000  AKR_J  156     74   298     504\n",
       "..               ...        ...    ...  ...    ...   ...     ...\n",
       "418       indelminer         10   LP_J   15  13401   137     483\n",
       "419       indelminer        100   LP_J   67  13349   137     483\n",
       "420       indelminer       1000   LP_J  189  13227   159     483\n",
       "421       indelminer      10000   LP_J  347  13069  1265     483\n",
       "422  manta_diploidSV         10   LP_J   60    422   873     483\n",
       "423  manta_diploidSV        100   LP_J  312    170  1052     483\n",
       "424  manta_diploidSV       1000   LP_J  337    145  1077     483\n",
       "425  manta_diploidSV      10000   LP_J  361    121  1293     483\n",
       "426         mistrvar         10   LP_J   18   5554    54     483\n",
       "427         mistrvar        100   LP_J   97   5475    59     483\n",
       "428         mistrvar       1000   LP_J  183   5389    82     483\n",
       "429         mistrvar      10000   LP_J  430   5142   789     483\n",
       "430           pindel         10   LP_J   43  76321    43     483\n",
       "431           pindel        100   LP_J  195  76169    45     483\n",
       "432           pindel       1000   LP_J  340  76024   131     483\n",
       "433           pindel      10000   LP_J  480  75884  4239     483\n",
       "434           popdel         10   LP_J    6    216   256     483\n",
       "435           popdel        100   LP_J  173     49   317     483\n",
       "436           popdel       1000   LP_J  210     12   336     483\n",
       "437           popdel      10000   LP_J  215      7   359     483\n",
       "438        rdxplorer         10   LP_J    4   2662    53     483\n",
       "439        rdxplorer        100   LP_J  132   2534    27     483\n",
       "440        rdxplorer       1000   LP_J  227   2439    59     483\n",
       "441        rdxplorer      10000   LP_J  393   2273   187     483\n",
       "442           smoove         10   LP_J   78    322   245     483\n",
       "443           smoove        100   LP_J  281    119   331     483\n",
       "444           smoove       1000   LP_J  315     85   348     483\n",
       "445           smoove      10000   LP_J  331     69   423     483\n",
       "446             gasv         10   LP_J    0  11605     9     483\n",
       "447            crest         10   LP_J    0    299  3517     483\n",
       "\n",
       "[448 rows x 7 columns]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculating precision, sensitivity, specificity, f-score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>n_true</th>\n",
       "      <th>nFN</th>\n",
       "      <th>sensitivity</th>\n",
       "      <th>precision</th>\n",
       "      <th>specificity</th>\n",
       "      <th>f-score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>36</td>\n",
       "      <td>389</td>\n",
       "      <td>206</td>\n",
       "      <td>504</td>\n",
       "      <td>468</td>\n",
       "      <td>0.071429</td>\n",
       "      <td>0.084706</td>\n",
       "      <td>0.346218</td>\n",
       "      <td>0.077503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>249</td>\n",
       "      <td>176</td>\n",
       "      <td>289</td>\n",
       "      <td>504</td>\n",
       "      <td>255</td>\n",
       "      <td>0.494048</td>\n",
       "      <td>0.585882</td>\n",
       "      <td>0.621505</td>\n",
       "      <td>0.536060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>269</td>\n",
       "      <td>156</td>\n",
       "      <td>299</td>\n",
       "      <td>504</td>\n",
       "      <td>235</td>\n",
       "      <td>0.533730</td>\n",
       "      <td>0.632941</td>\n",
       "      <td>0.657143</td>\n",
       "      <td>0.579117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>291</td>\n",
       "      <td>134</td>\n",
       "      <td>347</td>\n",
       "      <td>504</td>\n",
       "      <td>213</td>\n",
       "      <td>0.577381</td>\n",
       "      <td>0.684706</td>\n",
       "      <td>0.721414</td>\n",
       "      <td>0.626480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Tardis</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>51</td>\n",
       "      <td>245</td>\n",
       "      <td>284</td>\n",
       "      <td>504</td>\n",
       "      <td>453</td>\n",
       "      <td>0.101190</td>\n",
       "      <td>0.172297</td>\n",
       "      <td>0.536862</td>\n",
       "      <td>0.127500</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold strain  nTP  nFP  nTN  n_true  nFN  sensitivity  \\\n",
       "0  BioGraph*         10  AKR_J   36  389  206     504  468     0.071429   \n",
       "1  BioGraph*        100  AKR_J  249  176  289     504  255     0.494048   \n",
       "2  BioGraph*       1000  AKR_J  269  156  299     504  235     0.533730   \n",
       "3  BioGraph*      10000  AKR_J  291  134  347     504  213     0.577381   \n",
       "4     Tardis         10  AKR_J   51  245  284     504  453     0.101190   \n",
       "\n",
       "   precision  specificity   f-score  \n",
       "0   0.084706     0.346218  0.077503  \n",
       "1   0.585882     0.621505  0.536060  \n",
       "2   0.632941     0.657143  0.579117  \n",
       "3   0.684706     0.721414  0.626480  \n",
       "4   0.172297     0.536862  0.127500  "
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge['nFN']=df_merge['n_true']-df_merge['nTP']\n",
    "\n",
    "df_merge['sensitivity']=df_merge['nTP']/df_merge['n_true']\n",
    "df_merge['precision']=df_merge['nTP']/(df_merge['nTP']+df_merge['nFP'])\n",
    "df_merge['specificity']=df_merge['nTN']/(df_merge['nTN']+df_merge['nFP'])\n",
    "df_merge['f-score']=2*(df_merge['sensitivity']*df_merge['precision'])/(df_merge['sensitivity']+df_merge['precision']+0.00000001)\n",
    "df_merge.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nFN</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>297</td>\n",
       "      <td>2878</td>\n",
       "      <td>3502</td>\n",
       "      <td>1672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>1827</td>\n",
       "      <td>1348</td>\n",
       "      <td>1972</td>\n",
       "      <td>2249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>2004</td>\n",
       "      <td>1171</td>\n",
       "      <td>1795</td>\n",
       "      <td>2331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>2163</td>\n",
       "      <td>1012</td>\n",
       "      <td>1636</td>\n",
       "      <td>2717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Tardis</td>\n",
       "      <td>10</td>\n",
       "      <td>175</td>\n",
       "      <td>2048</td>\n",
       "      <td>3624</td>\n",
       "      <td>2449</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Tardis</td>\n",
       "      <td>100</td>\n",
       "      <td>1729</td>\n",
       "      <td>494</td>\n",
       "      <td>2070</td>\n",
       "      <td>2877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Tardis</td>\n",
       "      <td>1000</td>\n",
       "      <td>1924</td>\n",
       "      <td>299</td>\n",
       "      <td>1875</td>\n",
       "      <td>2942</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Tardis</td>\n",
       "      <td>10000</td>\n",
       "      <td>1983</td>\n",
       "      <td>240</td>\n",
       "      <td>1816</td>\n",
       "      <td>3248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>10</td>\n",
       "      <td>58</td>\n",
       "      <td>3484</td>\n",
       "      <td>3741</td>\n",
       "      <td>2026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>1883</td>\n",
       "      <td>1659</td>\n",
       "      <td>1916</td>\n",
       "      <td>2483</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          tool  threshold   nTP   nFP   nFN   nTN\n",
       "0    BioGraph*         10   297  2878  3502  1672\n",
       "1    BioGraph*        100  1827  1348  1972  2249\n",
       "2    BioGraph*       1000  2004  1171  1795  2331\n",
       "3    BioGraph*      10000  2163  1012  1636  2717\n",
       "4       Tardis         10   175  2048  3624  2449\n",
       "5       Tardis        100  1729   494  2070  2877\n",
       "6       Tardis       1000  1924   299  1875  2942\n",
       "7       Tardis      10000  1983   240  1816  3248\n",
       "8  breakdancer         10    58  3484  3741  2026\n",
       "9  breakdancer        100  1883  1659  1916  2483"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#count number of TP/FP/TN/FN for each tool at each threshold\n",
    "df_sum=df_merge.groupby(['tool','threshold'],as_index=False)['nTP','nFP','nFN','nTN'].sum()\n",
    "df_sum.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Stored 'df_sum' (DataFrame)\n"
     ]
    }
   ],
   "source": [
    "%store df_sum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sum.to_csv('../Data/analysis_files/df_sum.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_merge.to_csv('../Data/analysis_files/df_merged.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>n_true</th>\n",
       "      <th>nFN</th>\n",
       "      <th>sensitivity</th>\n",
       "      <th>precision</th>\n",
       "      <th>specificity</th>\n",
       "      <th>f-score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>36</td>\n",
       "      <td>389</td>\n",
       "      <td>206</td>\n",
       "      <td>504</td>\n",
       "      <td>468</td>\n",
       "      <td>0.071429</td>\n",
       "      <td>0.084706</td>\n",
       "      <td>0.346218</td>\n",
       "      <td>0.077503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>249</td>\n",
       "      <td>176</td>\n",
       "      <td>289</td>\n",
       "      <td>504</td>\n",
       "      <td>255</td>\n",
       "      <td>0.494048</td>\n",
       "      <td>0.585882</td>\n",
       "      <td>0.621505</td>\n",
       "      <td>0.536060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>269</td>\n",
       "      <td>156</td>\n",
       "      <td>299</td>\n",
       "      <td>504</td>\n",
       "      <td>235</td>\n",
       "      <td>0.533730</td>\n",
       "      <td>0.632941</td>\n",
       "      <td>0.657143</td>\n",
       "      <td>0.579117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>291</td>\n",
       "      <td>134</td>\n",
       "      <td>347</td>\n",
       "      <td>504</td>\n",
       "      <td>213</td>\n",
       "      <td>0.577381</td>\n",
       "      <td>0.684706</td>\n",
       "      <td>0.721414</td>\n",
       "      <td>0.626480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Tardis</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>51</td>\n",
       "      <td>245</td>\n",
       "      <td>284</td>\n",
       "      <td>504</td>\n",
       "      <td>453</td>\n",
       "      <td>0.101190</td>\n",
       "      <td>0.172297</td>\n",
       "      <td>0.536862</td>\n",
       "      <td>0.127500</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold strain  nTP  nFP  nTN  n_true  nFN  sensitivity  \\\n",
       "0  BioGraph*         10  AKR_J   36  389  206     504  468     0.071429   \n",
       "1  BioGraph*        100  AKR_J  249  176  289     504  255     0.494048   \n",
       "2  BioGraph*       1000  AKR_J  269  156  299     504  235     0.533730   \n",
       "3  BioGraph*      10000  AKR_J  291  134  347     504  213     0.577381   \n",
       "4     Tardis         10  AKR_J   51  245  284     504  453     0.101190   \n",
       "\n",
       "   precision  specificity   f-score  \n",
       "0   0.084706     0.346218  0.077503  \n",
       "1   0.585882     0.621505  0.536060  \n",
       "2   0.632941     0.657143  0.579117  \n",
       "3   0.684706     0.721414  0.626480  \n",
       "4   0.172297     0.536862  0.127500  "
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
